hystrix-contrib/hystrix-javanica/hystrix-javanica-core/src/main/java/com/netflix/hystrix/contrib/javanica/command/closure/BaseClosureFactory.java

```diff
@@ -60,27 +33,29 @@ public Closure createClosure(MetaHolder metaHolder, Method method, Object o, Obj
      * @return new {@link Closure} instance
      * @throws Exception
      */
-    Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
+   protected Closure createClosure(String rootMethodName, final Object closureObj) throws Exception {
         if (!isClosureCommand(closureObj)) {
             throw new RuntimeException(format(ERROR_TYPE_MESSAGE, rootMethodName,
                     getClosureCommandType().getName()).getMessage());
         }
         Method closureMethod = closureObj.getClass().getMethod(INVOKE_METHOD);
         return new Closure(closureMethod, closureObj);
     }
+    
+    protected abstract Object execute(T metaHolder, Method method, Object o, Object... args);
```

Comment:
I think I need to clarify a little bit here. Basically, this class `BaseClousureFactory` does only one thing - create a `Closure` based on the passed method arguments. Basically, it executes the passed `method` and wraps the result of the method execution in a `Closure` object. Here, the execution of method differs based on the implementation(aop-alliance, aspectj etc) and that is the reason I have defined it as a abstract method. So, `execute()` is very much part of `Closure` creation and is not used for any other purpose.